.github/workflows/build_and_tests.yml

```diff
@@ -18,42 +18,40 @@ jobs:
     - name: Clone
       uses: actions/checkout@v2
     - name: Speculos dependencies
-      run: sudo apt-get update && sudo apt-get install -y qemu-user-static
+      run: sudo apt-get update && sudo apt-get install -y qemu-user-static gcc-arm-linux-gnueabihf libvncserver-dev
     - name: Build & install
       run: |
         pip install --extra-index-url https://test.pypi.org/simple/ -U .[tests,all_backends]
         pip install -U click>=8
-    - name: Gather boilerplate NanoS to run tests against
-      uses: dawidd6/action-download-artifact@v2
-      with:
-        repo: LedgerHQ/app-boilerplate
-        name: boilerplate-app-nanos
-        workflow: ci-workflow.yml
-        path: tests/elfs/
-    - name: Renaming NanoS artifact
-      run: mv tests/elfs/app.elf tests/elfs/nanos.elf
-    - name: Gather boilerplate NanoX to run tests against
-      uses: dawidd6/action-download-artifact@v2
+    - name: Clone Speculos
+      uses: actions/checkout@v3
       with:
-        repo: LedgerHQ/app-boilerplate
-        name: boilerplate-app-nanox
-        workflow: ci-workflow.yml
-        path: tests/elfs/
-    - name: Renaming NanoS artifact
-      run: mv tests/elfs/app.elf tests/elfs/nanox.elf
-    - name: Gather boilerplate NanoS+ to run tests against
+        repository: LedgerHQ/speculos
+        ref: improve_client_events
+        path: speculos
+    - name: Install Speculos 
+      run: |
+        cd speculos
+        pip install .
+    - name: Gather boilerplate for Nano S/X/S+ to run tests against
       uses: dawidd6/action-download-artifact@v2
       with:
         repo: LedgerHQ/app-boilerplate
-        name: boilerplate-app-nanosp
+        name: apps
         workflow: ci-workflow.yml
         path: tests/elfs/
-    - name: Renaming NanoS artifact
-      run: mv tests/elfs/app.elf tests/elfs/nanosp.elf
+    - name: Renaming boilerplate binary artifacts
+      run: |
+        mv tests/elfs/boilerplate_nanos.elf tests/elfs/nanos.elf
+        mv tests/elfs/boilerplate_nanosp.elf tests/elfs/nanosp.elf
+        mv tests/elfs/boilerplate_nanox.elf tests/elfs/nanox.elf
     - name: Check the downloaded files
       run: tree .
     - name: Run tests and generate coverage
-      run: pytest tests/ --cov ragger --cov-report xml
+      run: |
+        pip show protobuf
```

Comment:
Do you think this should remain?

Reply:
@xchapron-ledger `pip show protobuf` can be removed but I'm not sure about the downgrade part (`pip install -I protobuf==3.20.0`). The version installed is not compatible with the proto files we have... Don't know when this problem appeared though.

Reply:
@fbeutin-ledger what do you think ?